dimagi_lake/aggregation/aggregation_helpers/child_care_monthly.py

```diff
@@ -0,0 +1,118 @@
+from dimagi_lake.aggregation.aggregation_helpers.base_helper import BaseAggregationHelper
+from spark_session_handler import SPARK
+from consts import SERVICE_ENROLLMENT_TABLE, FLWC_LOCATION_TABLE
+
+
+class ChildCareMonthlyAggregationHelper(BaseAggregationHelper):
+
+    @property
+    def source_tablename(self):
+        return f"{self.database_name}.raw_child_care_case"
+
+    @property
+    def member_case_table(self):
+        return f"{self.database_name}.raw_member_case"
+
+    @property
+    def location_table(self):
+        return f"{self.database_name}.{FLWC_LOCATION_TABLE}"
+
+    @property
+    def service_enrollment_table(self):
+        return f"{self.database_name}.{SERVICE_ENROLLMENT_TABLE}"
+
+    def aggregate(self):
+        df = self.preprocess()
+        df.show(10,False)
+        df = self.aggregate_data(df)
+        df.show()
+        return self.enforce_schema(df)
+
+    def preprocess(self):
+        return SPARK.sql(f"""
+        SELECT
+            case_id,
+            opened_on,
+            owner_id as flwc_id,
+            case_json__low_birth_weight as low_birth_weight,
+            case_json__breastfeeding_initiated as breastfeeding_initiated,
+            case_json__birth_weight as birth_weight,
+            case_references.referenced_id as member_case_id,
+            case_references.referenced_type as reference_type
+        FROM {self.source_tablename}
+        LATERAL VIEW explode(indices) as case_references
+        WHERE (
+            case_references.referenced_type='member' AND
+            case_references.identifier = 'parent' AND
+            month <='{self.month}' AND
+            (closed_on is null or closed_on >= '{self.month}')
+        )
+        """)
+
+    def aggregate_data(self, df):
+
+        age_in_months = f"datediff('{self.month}', member_case.case_json__member_dob)/30.4"
+        age_in_months_end = f"datediff(add_months('{self.month}', 1), member_case.case_json__member_dob)/30.4"
+
+        want_nutrition_services = (f"(service_enrollment.want_nutrition_services is distinct from 0 "
+                                   f" OR service_enrollment.registered_on>='{self.month}')")
+        want_growth_tracking_services = (f"(service_enrollment.want_growth_tracking_services is distinct from 0 "
+                                         f" OR service_enrollment.registered_on>='{self.month}')")
+        want_counselling_services = (f"(service_enrollment.want_counselling_services is distinct from 0 "
+                                     f"OR service_enrollment.registered_on>='{self.month}')")
+        born_in_month = (f"date_trunc('MONTH',case_json__member_dob)='{self.month}'")
+
+        df.createOrReplaceTempView("child_cases")
+
+        return SPARK.sql(f"""
+        SELECT
```

Comment:
this is an idle and very optional thought, but would it be worth moving these out into .sql files so we can get better dev tooling like syntax highlighting and such?

Reply:
I had the following thoughts while keeping these SQLs in the python file:
1. It makes it central. All aggregation logic of a table resides in a single file.
2. It also makes it easy in case we want to include some python/ pyspark logic in combination with the spark-SQL query all in one place.

I think I would prefer to keep it like this for now. Let's see how it goes. If we feel in the future it is messy, we should be able to separate them with LOW-Medium LOE

Reply:
:+1: those sound like good reasons to me